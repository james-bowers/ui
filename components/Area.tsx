import * as React from "react";
import styled, { css } from "./styled";

const sharedAreaStyle = css`
  width: 100%;
`;

const HeroArea = styled.div`
  background: ${props => props.theme.primaryColor};
  color: ${props => props.theme.primaryColorInverted};
  ${sharedAreaStyle}
`;

const StandardArea = styled.div`
  ${sharedAreaStyle}
`;

let StyledPadding = styled.div`
  &.padding_1 {
    padding: ${props => props.theme.spacing * 5}px;
  }

  &.padding_2 {
    padding: ${props => props.theme.spacing * 10}px;
  }

  &.padding_3 {
    padding: ${props => props.theme.spacing * 15}px;
  }

  &.padding_4 {
    padding: ${props => props.theme.spacing * 20}px;
  }

  &.padding_5 {
    padding: ${props => props.theme.spacing * 25}px;
  }
`;

export function Padding(props: {
  children: any;
  multiplier: 1 | 2 | 3 | 4 | 5;
}) {
  return (
    <StyledPadding className={`padding_${props.multiplier}`}>
      {props.children}
    </StyledPadding>
  );
}

export default (props: { hero?: boolean; children: any }) => {
  let Area = props.hero ? HeroArea : StandardArea;

  return <Area>{props.children}</Area>;
};
