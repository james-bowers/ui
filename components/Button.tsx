import * as React from "react";
import styled from "./styled";

type ButtonProps = {
  children: string;
  onClick?: () => Promise<boolean>;
};

let StyledButton = styled.button`
  font-family: ${props => props.theme.fontFamily};
  background: ${props => props.theme.accent};
  color: ${props => props.theme.accentInverted};
  padding: ${props => props.theme.spacing * 3}px;
  font-size: 1em;
  border: 0;
  border-radius: 5px;
  text-transform: uppercase;
  cursor: pointer;
`;

export default (props: ButtonProps) => (
  <StyledButton onClick={props.onClick}>{props.children}</StyledButton>
);
