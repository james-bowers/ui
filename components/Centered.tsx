import * as React from "react";
import styled from "./styled";

const StyledContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;

const StyledTopItem = styled.div`
  align-self: flex-start;
`;

const StyledBottomItem = styled.div`
  align-self: flex-end;
`;

export function Container(props: any) {
  return <StyledContainer {...props} />;
}

export function AtTop(props: any) {
  return <StyledTopItem {...props} />;
}

export function AtBottom(props: any) {
  return <StyledBottomItem {...props} />;
}
