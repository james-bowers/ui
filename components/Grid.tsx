import * as React from "react";
import styled from "./styled";

const StyledGridWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;

  &.width_equal > div.col {
    flex: 1;
  }

  ${[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
    .map(
      widthNum => `
    &.width_${widthNum} > div.col {
    width: ${(widthNum / 12) * 100}%;
  }`
    )
    .join("")}
`;

const StyledItem = styled.div``;

type Width = "equal" | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;

export let Container = (props: { children: any; width: Width }) => (
  <StyledGridWrapper className={`width_${props.width}`}>
    {props.children}
  </StyledGridWrapper>
);

export let Item = (props: { children: any }) => (
  <StyledItem className="col">{props.children}</StyledItem>
);
