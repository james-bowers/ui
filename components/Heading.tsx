import * as React from "react";
import styled, { css } from "./styled";

type HeadingProps = {
  as: "h1" | "h2" | "h3" | "h4" | "h5";
  children: any;
};

const sharedHeading = css`
  font-family: ${props => props.theme.fontFamily};
`;

const H1 = styled.h1`
  font-size: 3em;
  ${sharedHeading}
`;

const H2 = styled.h2`
  font-size: 2.5em;
  ${sharedHeading}
`;

const H3 = styled.h3`
  font-size: 2em;
  ${sharedHeading}
`;

const H4 = styled.h4`
  font-size: 1.5em;
  ${sharedHeading}
`;

export default (props: HeadingProps) => {
  if (props.as === "h2") return <H2>{props.children}</H2>;
  if (props.as === "h3") return <H3>{props.children}</H3>;
  if (props.as === "h4") return <H4>{props.children}</H4>;

  return <H1>{props.children}</H1>;
};
