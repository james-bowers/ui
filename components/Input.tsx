import * as React from "react";
import styled, { css } from "./styled";

type InputProps = {
  children: string;
  type: "text" | "number";
  name: string;
  onPrimary?: boolean;
};

let sharedInput = css`
  color: ${props => props.theme.primaryColorInverted};
  padding: ${props => props.theme.spacing * 2}%;
  border: 0;
  border-radius: 5px;
  min-width: 100%;
  font-family: ${props => props.theme.fontFamily};
  font-size: 0.8em;
`;
let StyledInputOnPrimary = styled.input`
  background: rgba(255, 255, 255, 0.09);
  ${sharedInput}
`;

let StyledInput = styled.input`
  background: ${props => props.theme.primaryColor};
  ${sharedInput}
`;

export const Input = (props: InputProps) => {
  let Component = props.onPrimary ? StyledInputOnPrimary : StyledInput;
  return (
    <Component
      name={props.name}
      type={props.type}
      defaultValue={props.children}
    />
  );
};
