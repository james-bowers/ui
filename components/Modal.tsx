import * as React from "react";
import styled, { keyframes } from "./styled";

type ModalProps = {
  children: any;
  open: boolean;
  onWrapperClicked?: Function;
};

let ModalWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(255, 255, 255, 0.5);
`;

let ModalContent = styled.div`
  max-width: 50%;
  max-height: 50%;
  background: ${props => props.theme.primaryColor};
  color: ${props => props.theme.primaryColorInverted};
`;

export default (props: ModalProps) => (
  <div>
    {props.open && (
      <ModalWrapper
        onClick={() => props.onWrapperClicked && props.onWrapperClicked()}
      >
        <ModalContent
          onClick={event => {
            event.preventDefault();
            event.stopPropagation();
            return false;
          }}
        >
          {props.children}
        </ModalContent>
      </ModalWrapper>
    )}
  </div>
);
