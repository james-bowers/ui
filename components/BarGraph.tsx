import * as React from "react";
import styled from "./styled";
import * as Grid from "./Grid";

export type DataPoint = {
  value: number;
};

export type BarGraphProps = {
  height: number;
  data: DataPoint[];
};

const StyledBar = styled.div`
  background: ${props => props.theme.accent};
  margin: 5px;
  align-self: flex-end;
  width: 100%;
  border-radius: 10px;
  max-width: 25px;
`;

const StyledBarContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
`;

function StyledDataPoint(props: { dataPoint: DataPoint }) {
  return <StyledBar style={{ height: `${props.dataPoint.value}%` }} />;
}

export default function BarGraph(props: BarGraphProps) {
  return (
    <div
      style={{
        height: `${props.height}px`
      }}
    >
      <Grid.Container width="equal">
        {props.data.map((dataPoint, _index) => (
          <Grid.Item key={_index}>
            <StyledBarContainer>
              <StyledDataPoint dataPoint={dataPoint} />
            </StyledBarContainer>
          </Grid.Item>
        ))}
      </Grid.Container>
    </div>
  );
}
