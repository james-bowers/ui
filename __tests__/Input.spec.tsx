import * as React from "react";
import { configure, shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

import * as Input from "../components/Input";

describe("<Input />", () => {
  test("renders Input", () => {
    const tree = shallow(
      <Input.Input type="text" name="example-input">
        example value
      </Input.Input>
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
