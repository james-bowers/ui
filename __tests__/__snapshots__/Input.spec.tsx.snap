// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders Input 1`] = `
<styled.input
  defaultValue="example value"
  name="example-input"
  type="text"
/>
`;
